src/modules/iotjs_module_process.c

```diff
@@ -290,7 +328,17 @@ iotjs_jval_t InitProcess() {
   // Set iotjs
   SetProcessIotjs(process);
 
-  SetProcessArgv(process);
+  bool wait_source =
+      iotjs_environment_config(iotjs_environment_get())->debugger_wait_source;
+
+  if (!wait_source) {
+    SetProcessArgv(process);
+  }
+
+  iotjs_jval_t wait_source_val = *iotjs_jval_get_boolean(wait_source);
+  iotjs_jval_set_property_jval(&process,
```

Comment:
Note: This must be updated if #1228 will land first. #1228 will change the prototype of `iotjs_jval_set_property_jval`.

Reply:
Thanks, I'll definitely update if your patch lands.